bot/pagination.py

```diff
@@ -87,6 +108,53 @@ def add_line(self, line: str = '', *, empty: bool = False) -> None:
             self._current_page.append('')
             self._count += 1
 
+        if remaining_words:
+            self._new_page()
+            self.add_line(remaining_words)
+
+    def _new_page(self) -> None:
+        self._linecount = 0
+        self._count = len(self.prefix) + 1
+        self.close_page()
+
+    def _split_remaining_words(self, line: str, max_chars: int) -> t.Tuple[str, t.Optional[str]]:
+        """
+        Internal: split a line into two strings -- reduced_words and remaining_words.
+
+        reduced_words: the remaining words in `line`, after attempting to remove all words that
+            exceed `max_chars` (rounding down to the nearest word boundary).
+
+        remaining_words: the words in `line` which exceed `max_chars`. This value is None if
+            no words could be split from `line`.
+
+        Return a tuple in the format (reduced_words, remaining_words).
+        """
+        reduced_words = []
+        remaining_words = []
+        # "(Continued)" is used on a line by itself to indicate the continuation of last page
+        continuation_header = "(Continued)\n-----------\n"
+        reduced_char_count = 0
+        is_full = False
```

Comment:
I'm on the fence about this boolean. I think naming boolean variables `is_adjective` or `has_noun` or whatever is a nice practice, but on the other hand, if this was just called `full` the code below would read `if not full`, which is lovely.

your call.

Reply:
With the ambiguity that Python has with `bool(obj)`, I put a lot of value is using the `is_`/`has_` prefix to directly communicate to readers (and myself, later down the road) that it's referring to a strictly boolean value. In this case, a name like `full` might be obvious and `is not full` sounds great, but I personally find it to be more explicit.

Reply:
then keep it as is. :+1: